Default tags for stacks

thiagokroger opened the pull request.

thiagokroger commented:
@irelandm @zimny 
I was taking a different approach before because I didn't think that the cost center could be like a configuration var.

This is a draft PR because I still have some questions related to that.

 - I believe the cost center is something that will vary according to the account, so is that ok to put it under `account.costCenter`?
 - Should the cost center be asked when making the default config? `make-default-config.ts`
 - In the case of CI where I can have multiple accounts, should the tags be created inside [this foreach](https://github.com/nearform/mira/blob/424cf5deb76be25a061f174a769f557fc5e486ee/src/cdk/constructs/cicd/index.ts#L97)?
 - @irelandm could you elaborate more on `expose a method to simply modify that list of default tags` is this something I can just validate using the json validator of the config file?

irelandm commented:
> * I believe the cost center is something that will vary according to the account, so is that ok to put it under `account.costCenter`?
We're assuming 1 cost centre per _application_, so from Mira point of view the same cost centre will cross all accounts for that application. So, answer is no.

> * Should the cost center be asked when making the default config? `make-default-config.ts`
We'll keep the wizard simple at this time. So not for now, no. 

> * In the case of CI where I can have multiple accounts, should the tags be created inside [this foreach](https://github.com/nearform/mira/blob/424cf5deb76be25a061f174a769f557fc5e486ee/src/cdk/constructs/cicd/index.ts#L97)?
Let's discuss (at sync)

> * @irelandm could you elaborate more on `expose a method to simply modify that list of default tags` is this something I can just validate using the json validator of the config file?
Per conversation with Piotr, the tags should go in a config file, so no need to provide a method for it now.

thiagokroger commented:
Ok, I just moved the cost center one level up

irelandm commented:
Thiago, please can you update the config doc page to include info on tags and how to set them

zimny commented on src/cdk/stack.ts:
```diff
@@ -31,6 +31,16 @@ export class MiraServiceStack extends cdk.Stack {
       })
+
+      Tag.add(this, 'StackName', this.stackName)
+      Tag.add(this, 'CreatedBy', account.env.account)
```
`account` will point to the current account number right? what is expected value here?

thiagokroger replied:
That's right, isn't that the owner of the stack?

irelandm replied:
Hmm, the account is where the stack is deployed, so we already know that information when looking at stacks.
The value of this field should be the userid of the user who created the stack.

irelandm commented on src/cdk/stack.ts:
```diff
@@ -31,6 +31,16 @@ export class MiraServiceStack extends cdk.Stack {
+
+      const costCenter = MiraConfig.getCostCenter()
+
+      if (costCenter) {
```
If I don't have a `costCenter` entry then I get 
>Failed: could not deploy the stack Error: Configuration property "costCenter" is not defined

Can you check the behaviour of the config module to make sure it is behaving as expected?

thiagokroger commented:
@irelandm updated with:
 - Fixed JSON Validator with accounts being object and adding cost center
 - Updated CreatedBy tag getting the logged in user
 - Cost center not required anymore
 - Added docs
